### PR TITLE
Update python FileAsset to accept os.PathLike in addition to str.

### LINKED ISSUE #2896: Support pathlib.Path in FileAsset
`pulumi.FileAsset` currently only supports strings, not `pathlib.Path` or anything else bearing `__fspath__()`.

### PR DESCRIPTION
**Pull request by campbellr**
This should fix #2896.

For the record I'm not very familiar with the codebase and it wasn't entirely clear to me what the tests are actually *doing*, so if the test changes I made are insufficient I'd be happy to fix them.

### CODE REVIEW

File: sdk/python/lib/pulumi/asset.py [hunk 1]
```diff
@@ -36,10 +37,11 @@ class FileAsset(Asset):
...
     """
-    def __init__(self, path: str) -> None:
-        if not isinstance(path, str):
-            raise TypeError("FileAsset path must be a string")
-        self.path = path
+
+    def __init__(self, path: Union[str, PathLike]) -> None:
+        if not isinstance(path, (str, PathLike)):
+            raise TypeError("FileAsset path must be a string or os.PathLike")
+        self.path = str(path)
```
Comment (lukehoban) [inline sdk/python/lib/pulumi/asset.py]: Minor - but I think technically it’s jot a guarantee that PathLike coerces correctly under `str()`. I suspect this should be `os.fspath()`?

Comment (campbellr) [reply sdk/python/lib/pulumi/asset.py]: Oh thanks, I didn't even think about that since `pathlib.Path` they're the same...

I'll update this.

Comment (campbellr) [reply sdk/python/lib/pulumi/asset.py]: Fixed.

**Comment by pgavlin** [comment]
Running a buddy build of the tests now.

**Comment by pgavlin** [comment]
We're good. Going to go ahead and merge after resolving conflicts.

**Pull request merged**